src/shared/proto/BUILD

```diff
@@ -0,0 +1,18 @@
+load("@rules_proto//proto:defs.bzl", "proto_library")
+
+package(default_visibility = ["//visibility:public"])
+
+proto_library(
+    name = "tbots_proto",
+    srcs = [
+        "tbots_robot_msg.proto",
+        "tbots_robot_status_2014.proto",
```

Comment:
We should be consistent about how we label the old robots. In the firmware I've labelled them as `legacy_robot_stm32f4`, but I think I prefer the naming you have here.

All that said, why are there different proto's for the old and new robots? Could we have a single proto with optional fields for the fields that the old robots don't supply?

Reply:
if we had optional fields for all the fields that are different between the new and old robots, then SensorFusion would have to check before using each one of them, instead of just checking once for the version of the proto.

Reply:
What would the API to the filters be in that case? For example, assuming we move forward with EKFs, they can work with as much or as little data as we have. Would we need two EKFs versions to support each type of proto?

Reply:
I'd imagine we'd have functions like filterRobotStatus(Tbots_RobotStatus2014 status) and filterRobotStatus(Tbots_FrankieStatus status)

Reply:
But how would that look internally? Assuming the status messages share some common data (battery, cap voltage, etc), not that these are used in filtering, this seems like it would cause duplication in the filters.

For the sake of argument, what it the only difference between 2 robot generations was that one had an IMU and the other did not?

@garethellis0 for how / if this matters for standard EKF structure.

Reply:
Hmm. Don't know that I'm a huge fan of just checking the type and just handling different types completely differently. This proto shouldn't have any concept of how the EKF works, it should just hold information that the EKF uses, like encoder counts, if that helps answer your question?

Reply:
Yea I was thinking of how "bad" it is for the EKF to work around the proto. Would a single EKF be able to support two message types just with a different interface, or would we need 1 EKF per type? Agree that our proto design shouldn't make assumptions about how it will be used in filtering.

Reply:
I'm a bit confused about what you're asking here. The EKF shouldn't be related at all to the proto, both robots will supply encoder readings in the end, so there's no difference in the information that makes it to the EKF.

Reply:
I'm also confused, lets discuss in the meeting today.

Reply:
wow look at me replying to comments @garethellis0